comprl/server/managers.py

```diff
@@ -282,20 +320,132 @@ def remove(self, player_id: PlayerID) -> None:
         Args:
             player_id (PlayerID): The ID of the player to be removed.
         """
-        self.queue = [p for p in self.queue if (p != player_id)]
+        self._queue = [
+            (p_id, u_id, mu, sigma, time)
+            for p_id, u_id, mu, sigma, time in self._queue
+            if (p_id != player_id)
+        ]
 
     def update(self):
         """
         Updates the matchmaking manager.
         """
-        pass
+
+        if len(self._queue) < self._min_players_waiting():
+            return
+
+        for i in range(len(self._queue)):
+            for j in range(i + 1, len(self._queue)):
+                # try to match all players against each other
+                if self._try_start_game(self._queue[i], self._queue[j]):
+                    return
```

Comment:
I think you need a `break` here, instead of the `return`, if you want to start more than one game. In that case you should also test if there are enough players waiting to match more players. But you have to be careful, because you're iterating over a list that you are removing items from. This means you should check in every iteration of the outer loop, that there is no index out of bounds.

Reply:
Ahh good catch, thx. I changed the `_update()` function to have an optional start index, so we can recursively continue searching for matches,